stacker/actions/build.py

```diff
@@ -192,18 +193,37 @@ def _launch_stack(self, stack, **kwargs):
             provider_stack = None
 
         old_status = kwargs.get("status")
-        if provider_stack and old_status == SUBMITTED:
-            logger.debug(
-                "Stack %s provider status: %s",
-                stack.fqn,
-                self.provider.get_stack_status(provider_stack),
-            )
-            if self.provider.is_stack_completed(provider_stack):
-                submit_reason = getattr(old_status, "reason", None)
-                return CompleteStatus(submit_reason)
-            elif self.provider.is_stack_in_progress(provider_stack):
-                logger.debug("Stack %s in progress.", stack.fqn)
-                return old_status
+
+        logger.debug("Stack %s old_status=%r  provider=%r",
+                     stack.fqn, old_status, provider_stack)
+
+        if provider_stack:
```

Comment:
This logic seems scary - especially using `CompleteStatus` to mean that a stack is in rollback, since it would continue  executing other stacks while the stack was in a rollback state, and in most cases I think people would want to stop immediately rather than continue.